src/components/TeacherComponents/OnboardingForm/onboardingActions/collectOnboardingTrackingProps.ts

```diff
@@ -0,0 +1,143 @@
+    componentType: "continue_button",
+    eventVersion: "2.0.0",
+    analyticsUseCase: "Teacher",
+    userId_: user.id,
+    signUpDate,
+    userRoleType: pickUserRole(data),
+    teacherSchoolUrn: pickSchoolUrn(data),
+    teacherSchoolManualEntryDetails:
+      "manualSchoolName" in data
+        ? {
+            schoolNameText: data.manualSchoolName,
+            schoolAddressText: data.schoolAddress,
+          }
+        : null,
+    userDefinedRole: pickUserDefinedRole(data),
+    userDetailsLastModifiedDate,
+    userAccountVerificationStatus: pickVerificationStatus(user),
+  };
+}
+
+function isValidRoleOption(value: unknown): value is OnboardingRoleValue {
+  return Object.values(onboardingRoleOptions).includes(
+    value as OnboardingRoleValue,
+  );
+}
+
+function pickUserRole(data: OnboardingFormProps): UserRoleTypeValueType | null {
+  if ("role" in data && isValidRoleOption(data.role)) {
+    return mapOnboardingUserRole(data.role);
+  }
+
+  if ("worksInSchool" in data) {
+    return data.worksInSchool ? "Teacher" : "Non Teacher";
+  }
+
+  return null;
+}
+
+function pickVerificationStatus(
+  user: UserResource,
+): UserAccountVerificationStatusValueType {
+  const verificationStatuses = user.emailAddresses.map(
+    (address) => address.verification.status,
+  );
+
+  switch (true) {
+    case verificationStatuses.includes("verified"):
+      return "verified";
+    case verificationStatuses.includes("failed"):
+      return "disabled";
+    default:
+      return "awaiting-verification";
+  }
+}
+
+function mapOnboardingUserRole(
+  roleValue: OnboardingRoleValue,
+): UserRoleTypeValueType {
+  switch (roleValue) {
+    case onboardingRoleOptions["adult-helping-child"]:
+      return "Adult (Other)";
+    case onboardingRoleOptions["mat-or-lea"]:
+      return "LEA/MAT Professional";
+    case onboardingRoleOptions["private-tutor"]:
+      return "Private Tutor";
+    case onboardingRoleOptions["nonprofit"]:
+      return "Educational Not-for-Profit Professional";
+    case onboardingRoleOptions["teacher-training"]:
+      return "Trainee Teacher";
+    case onboardingRoleOptions["teacher-trainer"]:
+      return "Teacher Training Professional";
+    case onboardingRoleOptions["other"]:
+      return "Other";
+  }
+}
+
+function pickUserDefinedRole(data: OnboardingFormProps) {
+  if ("role" in data && data.role === onboardingRoleOptions.other) {
+    return data.other ?? null;
+  }
+
+  return null;
+}
+
+function pickSchoolUrn(data: OnboardingFormProps) {
+  if ("school" in data) {
+    return extractUrn(data.school) ?? DUMMY_URN;
+  }
+
+  if ("manualSchoolName" in data) {
+    return DUMMY_URN;
+  }
+
+  return null;
+}
+
+/**
+ * Pulls the URN/IRN from the school picker output
+ *
+ * English and Welsh URNs are 6 digits
+ * NI IRNs include a hypen `999-9999`
+ * Scottish URNs are 7 digits
+ */
+function extractUrn(school: string) {
+  return /^\d{7}|^\d{6}|^\d{3}-\d{4}/.exec(school)?.at(0);
```

Comment:
❗this has just made me realise there's a bug in the current school picker since adding NI schools! We get the URN by splitting at the hyphen... 🙃

Reply:
Always happy to cause issues

Reply:
To add to this. Do you know if we have LEA/Estab (for English/Welsh schools) as a possible field in the Aircury data. I think the reason for the oddness here is that NI/Scottish IDs are actually just representations of a `/\d{3}\d{4}/` code where the first three digits point to a local authority and the latter four point to an Establishment. This ID is typically a lot more stable than URN (which changes whenever schools convert to academy amongst other things). I wonder if would be good long term to only track URN for English Schools and to grab the 7 digit LEA/Estab code for all schools across the UK.

Reply:
@ewan-wakeman AIrcury get their data for English schools from GIS [which does appear to offer the LAESTAB](https://get-information-schools.service.gov.uk/Establishments/Search?SelectedTab=Establishments&SearchType=Location&SearchType=Location&LocationSearchModel.Text=London%2c+London&LocationSearchModel.AutoSuggestValue=51.50014877319336%2c+-0.12624000012874603&OpenOnly=true&b=1&b=4), but Aircury don't currently include it in their API responses.

E.g. this is what we currently get.
```json
{"urn":"100000","la":"City of London","name":"The Aldgate School","postcode":"EC3A 5DE","fullInfo":"100000, City of London, The Aldgate School, EC3A 5DE","status":"Open"}
``` 

It would likely be a chargeable piece of work for it to be added so that we can access it.

Reply:
Ok cool, it's not urgent at all, just might be one to consider longer term I suppose, seeing as how this seems to align more neatly with Scotland and NI. In the meantime we can deal with this nuance further down in the pipeline and just accept some slight more tricky regex for now.